tools/rosgraph/src/rosgraph/roslogging.py

```diff
@@ -122,10 +125,18 @@ def configure_logging(logname, level=logging.INFO, filename=None, env=None):
     
     # pass in log_filename as argument to pylogging.conf
     os.environ['ROS_LOG_FILENAME'] = log_filename
-    # #3625: disabling_existing_loggers=False
-    logging.config.fileConfig(config_file, disable_existing_loggers=False)
+    if config_file.endswith('.conf'):
+        # #3625: disabling_existing_loggers=False
+        logging.config.fileConfig(config_file, disable_existing_loggers=False)
+    elif config_file.endswith('.yaml'):
+        with open(config_file) as f:
+            dict_conf = yaml.load(f)
+            dict_conf.setdefault('version', 1)
+            logging.config.dictConfig(dict_conf)
+
```

Comment:
Currently this block doesn't have an `else` part. Also if the user was using a different filename before it might not work with the modified code anymore. Therefore I would suggest the logic to be updated to:

```
if yaml:
    # do yaml stuff
else:
    # do stuff as before
```